apps/api/src/controller/dework/index.ts

```diff
@@ -16,75 +17,120 @@ export const DeworkController = async (
   response: express.Response,
   next: express.NextFunction
 ) => {
-  console.log(request.body);
   try {
     if (!request?.body) {
       throw new Error('Body not defined');
     }
 
-    if (!request?.body?.address) {
-      throw new Error(`No address in body`);
+    if (!request?.body?.claimedCredentialId) {
+      throw new Error(`No claimedCredentialId in body`);
     }
 
-    const { address } = request.body;
+    const { claimedCredentialId } = request.body;
     const { wallet, ethProvider } = await connect();
 
     // Log in with wallet to Ceramic DID
     const Issuer = new krebit.core.Krebit({
       wallet,
       ethProvider,
       address: wallet.address,
-      ceramicUrl: SERVER_CERAMIC_URL
+      ceramicUrl: SERVER_CERAMIC_URL,
+      litSdk: LitJsSdk
     });
     const did = await Issuer.connect();
     console.log('DID:', did);
 
-    // Connect to dework and get reputation from address
-    const dework = await getDeworkUser({ address });
-    console.log('Importing from Dework:', dework.address);
+    const claimedCredential = await Issuer.getCredential(claimedCredentialId);
 
-    const expirationDate = new Date();
-    const expiresYears = parseInt(SERVER_EXPIRES_YEARS, 10);
-    expirationDate.setFullYear(expirationDate.getFullYear() + expiresYears);
-    console.log('expirationDate: ', expirationDate);
+    console.log('Verifying Dework with claimedCredential: ', claimedCredential);
 
-    if (!dework.tasks || dework.tasks.length === 0) {
-      throw new Error('Missing tasks');
+    // Check self-signature
+    console.log('checkCredential: ', Issuer.checkCredential(claimedCredential));
+
+    // get the claimValue
+    let claimValue = null;
+    //Decrypt
+    if (claimedCredential.credentialSubject.encrypted === 'lit') {
+      claimValue = await Issuer.decryptCredential(claimedCredential);
+    } else {
+      claimValue = JSON.parse(claimedCredential.credentialSubject.value);
+      console.log('Claim value: ', claimValue);
     }
+    const publicClaim: boolean =
+      claimedCredential.credentialSubject.encrypted === 'none';
+
+    //Tasks for a specific  Organization
+    if (
+      claimedCredential?.credentialSubject?.type === 'DeworkCompletedTasksGT10'
+    ) {
+      // Connect to dework and get reputation from address
+
+      const dework = await getDeworkUser({
+        address: claimedCredential.credentialSubject.ethereumAddress
+      });
+      console.log('Importing from Dework:', dework);
+
+      const orgTask = dework.tasks.filter(task => {
```

Comment:
this can be just: `const orgTask = dework.tasks.filter(task => task.workspace.organization.name.toLowerCase() ===
          claimValue.entity.toLowerCase());`

Reply:
Add ? to avoid any problem